g1ant-language/refining-your-code.md

```diff
@@ -0,0 +1,94 @@
+## Refining Your Code
+
+The code is never good enough. Of course, the golden rule says that if it works, don’t fix it — and we couldn’t agree more. But when you are practicing with programming, refinements are the part of the learning process. You learn new things and you can look back at your previous works to see if it’s possible to use your latest knowledge to optimize the code, make it clearer and more efficient.
+
+In the past two exercises, you used some basic commands and logic. In the first code, you can notice that there are many lines that could be reduced to just one, using a list variable and a loop:
+
+```G1
+excel.getvalue row ♥rownumber colname a result ♥cell1
+excel.getvalue row ♥rownumber colname b result ♥cell2
+excel.getvalue row ♥rownumber colname c result ♥cell3
+excel.getvalue row ♥rownumber colname d result ♥cell4
+excel.getvalue row ♥rownumber colname e result ♥cell5
+excel.getvalue row ♥rownumber colname f result ♥cell6
+```
+
+There is the `♥rownumber` variable which takes its value from the `for` loop, so it stays the same in all 6 rows of the procedure. How can you make the other two arguments change incrementally in the same line as the first one? By employing a list variable and another loop.
+
+Let’s start with the `♥cell` variable. Instead of creating a new variable for each cell, you could just create a list variable and add each cell’s content to this list as a value of a subsequent element. The `♥cell` list variable would look like this: `⟦cell1❚cell2❚cell3❚cell4❚cell5❚cell6⟧`. In order to move through the list elements, you would use the good old `foreach` loop or ordinary `for` loop and incremental indexing in the list variable.
```

Comment:
why did you create a list  ♥cell = ⟦cell1❚cell2❚cell3❚cell4❚cell5❚cell6⟧? what if you had 100 elements in that list? you should create an empty list instead:
`list.create type integer result ♥cells`
also I don't get why you inserted it inside `⟦⟧`. These characters are used to force parsing some value to some specified type for example: 
♥values = ⟦list:integer⟧0❚3❚10